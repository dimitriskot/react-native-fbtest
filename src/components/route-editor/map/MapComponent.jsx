import React from "react";
import PropTypes from "prop-types";
import { compose, withProps } from "recompose";
import {
  withScriptjs,
  withGoogleMap,
  GoogleMap,
  Marker,
  DirectionsRenderer
} from "react-google-maps";

const MapComponent = compose(
  withProps({
    googleMapURL: "https://maps.googleapis.com/maps/api/js?key=AIzaSyDM_DVnbVRiQLfPIOwyDFbwg5X_HIG62_Y",
    loadingElement: <div className={"loadingElement"} style={{ height: "100%" }} />,
    containerElement: <div className={"containerElement"} />,
    mapElement: <div className={"mapElement"} />
  }),
  withScriptjs,
  withGoogleMap
)((props) => {
  const {
    zoom,
    center,
    markers,
    directions,
    getMapRef,
    getMapCenter
  } = props;

  const markerList = markers.map(({ id, position, draggable, handleDragEnd }) => (
    <Marker
      key={id}
      position={position}
      draggable={draggable}
      onDragEnd={handleDragEnd.bind(this, id)}
    />
  ));

  return (
    <GoogleMap
      ref={getMapRef}
      defaultZoom={zoom}
      defaultCenter={center}
      onCenterChanged={getMapCenter}
    >
      {markerList}
      {directions && (
        <DirectionsRenderer
          directions={directions}
          options={{
            suppressMarkers: true,
            preserveViewport: true
          }}
        />
      )}
    </GoogleMap>
  );
});

MapComponent.propTypes = {
  zoom: PropTypes.number,
  center: PropTypes.object,
  directions: PropTypes.object,
  getMapRef: PropTypes.func,
  getMapCenter: PropTypes.func,
  markers: PropTypes.array,
  onDragEnd: PropTypes.func
};

export default MapComponent;
