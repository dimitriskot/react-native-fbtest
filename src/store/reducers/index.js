import { combineReducers } from "redux";
import { routerReducer } from "react-router-redux";
import productsReducer from "./products";

const fbTestApp = combineReducers({
  routing: routerReducer,
  productsReducer
});

export default fbTestApp;
