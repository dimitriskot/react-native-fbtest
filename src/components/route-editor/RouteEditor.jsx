import React from "react";
import MediaQuery from "react-responsive";
import classNames from "classnames";
import BackButton from "../common/BackButton";
import MapForm from "../../containers/route-editor/Form";
import MapComponent from "../../containers/route-editor/Map";

export default class RouteEditor extends React.Component {

  state = { isRouteListOpen: false };

  toggleMapForm = (e) => {
    e.preventDefault();
    const { isRouteListOpen } = this.state;
    this.setState({ isRouteListOpen: !isRouteListOpen });
  }

  render() {
    const { isRouteListOpen } = this.state;

    return (
      <div className={"route-editor"}>
        <BackButton />
        <div className={"route-editor__content"}>
          <MediaQuery maxWidth={768}>
            <button
              className={classNames("tab-button", isRouteListOpen && "tab-button--close")}
              onClick={this.toggleMapForm}
            >
              {isRouteListOpen ? "Закрыть" : "Маршрут"}
            </button>
            {isRouteListOpen && (<MapForm />)}
          </MediaQuery>
          <MediaQuery minWidth={769}>
            <MapForm />
          </MediaQuery>
          <MapComponent />
        </div>
      </div>
    );
  }

}
