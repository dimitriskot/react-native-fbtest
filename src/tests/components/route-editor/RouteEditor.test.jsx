import React from "react";
import { Provider } from "react-redux";
import MediaQuery from "react-responsive";
import { MemoryRouter, Route } from "react-router-dom";
import { mount } from "enzyme";
import store from "../../../store";
import RouteEditor from "../../../components/route-editor/RouteEditor";
import MapComponent from "../../../containers/route-editor/Map";

describe("RouteEditor component", () => {

  it("render RouteEditor", () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/", "/market", "/route-editor"]} initialIndex={1}>
          <Route path={"/route-editor"}>
            <RouteEditor />
          </Route>
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.render()).toMatchSnapshot();
    expect(wrapper.find(MediaQuery)).toHaveLength(2);
    expect(wrapper.find(MapComponent)).toHaveLength(1);
  });

});
