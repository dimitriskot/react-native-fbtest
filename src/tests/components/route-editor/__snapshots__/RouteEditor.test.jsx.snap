// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteEditor component render RouteEditor 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "route-editor",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "back-button",
          "href": "/",
        },
        "children": Array [
          Object {
            "data": "Назад",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "route-editor__content",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "loadingElement",
                "style": "height: 100%;",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "style": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "href": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "href": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "route-editor__content",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "loadingElement",
              "style": "height: 100%;",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "back-button",
            "href": "/",
          },
          "children": Array [
            Object {
              "data": "Назад",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
